src/utp_socket_manager.cpp

```diff
@@ -255,10 +255,10 @@ namespace libtorrent
 
 	void utp_socket_manager::writable()
 	{
-		std::vector<utp_socket_impl*> stalled_sockets;
-		m_stalled_sockets.swap(stalled_sockets);
-		for (std::vector<utp_socket_impl*>::iterator i = stalled_sockets.begin()
-			, end(stalled_sockets.end()); i != end; ++i)
+		m_stalled_sockets_tock.clear();
+		m_stalled_sockets.swap(m_stalled_sockets_tock);
+		for (socket_vector_t::iterator i = m_stalled_sockets_tock.begin()
```

Comment:
this should be a range-based for-loop. Something like: ``for (auto const& s : m_stalled_sockets_tock)``.